ext/spl/spl_iterators.stub.php

```diff
@@ -53,7 +53,7 @@ public function getChildren();
 
 class RecursiveIteratorIterator implements OuterIterator
 {
-    public function __construct(Traversable $iterator, int $mode = self::LEAVES_ONLY, int $flags = 0) {}
+    public function __construct(Traversable $iterator, int $mode = RecursiveIteratorIterator::LEAVES_ONLY, int $flags = 0) {}
```

Comment:
I think it would be better to address this in `gen_stubs`, by resolving `self` to explicit class reference.

(Otherwise we should probably add a check to gen_stubs that forbids use of self.)

Reply:
OK, good idea!

Reply:
Finally, I settled on forbidding the usage of self, since resolving it to the explicit class name would have been much more difficult to achieve (at least I didn't find an easy way).